codec-http2/src/test/java/io/netty/handler/codec/http2/UniformStreamByteDistributorTest.java

```diff
@@ -0,0 +1,306 @@
+        updateStream(STREAM_B, CHUNK_SIZE, true);
+        updateStream(STREAM_A, CHUNK_SIZE, true);
+
+        // Only write 3 * chunkSize, so that we'll only write to the first 3 streams.
+        int written = 3 * CHUNK_SIZE;
+        assertTrue(write(written));
+        assertEquals(CHUNK_SIZE, captureWrite(STREAM_A));
+        assertEquals(CHUNK_SIZE, captureWrite(STREAM_B));
+        assertEquals(CHUNK_SIZE, captureWrite(STREAM_C));
+        assertEquals(0, captureWrite(STREAM_D));
+        verifyNoMoreInteractions(writer);
+
+        reset(writer);
+
+        // Now write again and verify that the last stream is written to.
+        assertFalse(write(CHUNK_SIZE));
+        assertEquals(0, captureWrite(STREAM_A));
+        assertEquals(0, captureWrite(STREAM_B));
+        assertEquals(0, captureWrite(STREAM_C));
+        assertEquals(CHUNK_SIZE, captureWrite(STREAM_D));
+        verifyNoMoreInteractions(writer);
+    }
+
+    @Test
+    public void streamWithMoreDataShouldBeEnqueuedAfterWrite() throws Http2Exception {
+        // Give the stream a bunch of data.
+        updateStream(STREAM_A, 2 * CHUNK_SIZE, true);
+
+        // Write only part of the data.
+        assertTrue(write(CHUNK_SIZE));
+        assertEquals(CHUNK_SIZE, captureWrite(STREAM_A));
+        assertEquals(0, captureWrite(STREAM_B));
+        assertEquals(0, captureWrite(STREAM_C));
+        assertEquals(0, captureWrite(STREAM_D));
+        verifyNoMoreInteractions(writer);
+
+        reset(writer);
+
+        // Now write the rest of the data.
+        assertFalse(write(CHUNK_SIZE));
+        assertEquals(CHUNK_SIZE, captureWrite(STREAM_A));
+        assertEquals(0, captureWrite(STREAM_B));
+        assertEquals(0, captureWrite(STREAM_C));
+        assertEquals(0, captureWrite(STREAM_D));
+        verifyNoMoreInteractions(writer);
+    }
+
+    /**
+     * In this test, we verify that each stream is allocated a minimum chunk size. When bytes
+     * run out, the remaining streams will be next in line for the next iteration.
+     */
+    @Test
+    public void firstEnqueue() throws Http2Exception {
+        // Re-assign weights.
+        setPriority(STREAM_A, 0, (short) 50, false);
+        setPriority(STREAM_B, 0, (short) 200, false);
+        setPriority(STREAM_C, STREAM_A, (short) 100, false);
+        setPriority(STREAM_D, STREAM_A, (short) 100, false);
+
+        int chunkSize = DEFAULT_MIN_ALLOCATION_CHUNK;
+        updateStream(STREAM_A, chunkSize, true);
+        updateStream(STREAM_B, chunkSize, true);
+        updateStream(STREAM_C, chunkSize, true);
+        updateStream(STREAM_D, chunkSize, true);
+
+        // Only write 3 * chunkSize, so that we'll only write to the first 3 streams.
+        int written = 3 * chunkSize;
+        assertTrue(write(written));
+        List<Integer> streamsNotWritten = verifyWrite(written, 3, STREAM_A, STREAM_B, STREAM_C, STREAM_D);
+        assertEquals(1, streamsNotWritten.size());
```

Comment:
should we also assert which stream we expect not to be written?  is it `stream_a`?